### ISSUE TITLE
Upgrade journalbeat to use beats 5.0.0 framework and release it as 5.0.0

### ISSUE THREAD

**Issue by mheese** [mheese/journalbeat#38]
We need to update journalbeat to use the v5.0.0 of libbeat. At the same time we should start making releases which follow the same semver that all elastic stack products use now.

**Comment by nicorevin** [comment]
Was planning to do this over the weekend and test it out on Monday-Tuesday.

**Comment by mheese** [comment]
@nicorevin, sure I'll assign it to you then 👍 ... also, could you refactor the `vendor.json` and add the libbeat part as a tree dependency? Remove everything related from the vendor file first and delete all source files from the vendor folder, and then do a `govendor fetch -tree github.com/elastic/beats/libbeat@v5.0.0` which should do the trick.

**Comment by a86c6f7964** [comment]
any news on this?

**Comment by nicorevin** [comment]
@a86c6f7964 ETA - 2 weeks (I hope no more than that due to various distraction factors...)

**Comment by mikeatm** [comment]
Does the current version of journalbeat  push to logstash 5.x?  im trying to get it working but im not sure whether 
i should expect it to.

**Comment by nicorevin** [comment]
@mikeatm it's 50/50 since JB is based on the v5.0.0-alpha4. And they've reworked a big lot since then.
@mheese @a86c6f7964 so I hope I'm on the finish line with v5.1.1 and upstream go-systemd.

**Comment by nicorevin** [comment]
@mheese is there any benefit from using `fetch -tree`? 
It:
- makes vendor folder beefier due to fetching the entire `beats`, 
- turns `govendor list` into a mess if fetching the needed deps one by one with `-tree`,
- can potentially break stuff since it does not respect version tag when fetching part of subpackages and deps,
- throws errors when trying to freeze versions of the packages.

No `-tree`:
```
 v  github.com/Shopify/sarama                                               
 v  github.com/coreos/go-systemd/sdjournal                          v14     v14
 v  github.com/coreos/pkg/dlopen                                            
 v  github.com/davecgh/go-spew/spew                                         
 v  github.com/eapache/go-resiliency/breaker                                
 v  github.com/eapache/go-xerial-snappy                                     
 v  github.com/eapache/queue                                                
 v  github.com/elastic/beats/libbeat/beat                           v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/cfgfile                        v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/common                         v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/common/dtfmt                   v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/common/fmtstr                  v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/common/jsontransform           v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/common/op                      v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/common/streambuf               v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/logp                           v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs                        v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/console                v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/elasticsearch          v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/fileout                v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/kafka                  v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/logstash               v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/mode                   v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/mode/lb                v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/mode/modeutil          v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/mode/single            v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/outil                  v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/redis                  v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/outputs/transport              v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/paths                          v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/processors                     v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/processors/actions             v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/processors/add_cloud_metadata  v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/publisher                      v5.1.1  v5.1.1
 v  github.com/elastic/beats/libbeat/service                        v5.1.1  v5.1.1
 v  github.com/elastic/beats/metricbeat/schema                      v5.1.1  v5.1.1
 v  github.com/elastic/beats/metricbeat/schema/mapstriface          v5.1.1  v5.1.1
 v  github.com/elastic/go-lumber/client/v2                                  
 v  github.com/elastic/go-lumber/log                                        
 v  github.com/elastic/go-lumber/protocol/v2                                
 v  github.com/elastic/go-ucfg                                              
 v  github.com/elastic/go-ucfg/cfgutil                                      
 v  github.com/elastic/go-ucfg/flag                                         
 v  github.com/elastic/go-ucfg/internal/parse                               
 v  github.com/elastic/go-ucfg/json                                         
 v  github.com/elastic/go-ucfg/yaml                                         
 v  github.com/garyburd/redigo/internal                                     
 v  github.com/garyburd/redigo/redis                                        
 v  github.com/golang/snappy                                                
 v  github.com/joeshaw/multierror                                           
 v  github.com/klauspost/compress/flate                                     
 v  github.com/klauspost/compress/zlib                                      
 v  github.com/klauspost/cpuid                                              
 v  github.com/klauspost/crc32                                              
 v  github.com/nranchev/go-libGeoIP                                         
 v  github.com/pierrec/lz4                                                  
 v  github.com/pierrec/xxHash/xxHash32                                      
 v  github.com/pkg/errors                                                   
 v  github.com/rcrowley/go-metrics                                          
 v  github.com/rcrowley/go-metrics/exp                                      
 v  github.com/satori/go.uuid                                               
 v  golang.org/x/net/proxy                                                  
 v  golang.org/x/sys/windows                                                
 v  golang.org/x/sys/windows/svc                                            
 v  golang.org/x/sys/windows/svc/debug                                      
 v  gopkg.in/yaml.v2                                                        
pl  github.com/mheese/journalbeat                                    
 l  github.com/mheese/journalbeat/beater                             
 l  github.com/mheese/journalbeat/config      
```

With `-tree`:
```
 vt github.com/elastic/beats/libbeat/logp                           v5.1.1  v5.1.1
 vt github.com/elastic/beats/libbeat/mock                                   
 vt github.com/elastic/beats/libbeat/outputs                                

$ govendor fetch -v -tree github.com/elastic/beats/libbeat/outputs@v5.1.1
Error: Cannot add package "github.com/elastic/beats/libbeat/outputs" which is already found in sub-tree ["github.com/mheese/journalbeat/vendor/github.com/elastic/beats/libbeat"]
```

**Comment by mheese** [comment]
hm ... it worked quite well the last times I used it ... but you got a point. keep it pragmatic: if it gets in the way, don't use it.

I'm wondering about this though: `github.com/coreos/go-systemd/sdjournal` . There should *not* be any dependency to the CoreOS repo from our code ... as we currently rely on my fork of `go-systemd`

**Comment by nicorevin** [comment]
we won't rely on the fork in 5.1.1 ;)

**Comment by mheese** [comment]
@nicorevin we have to ... this is specific to journalbeat and has nothing to do with beats, but with the `go-systemd` library from CoreOS being incomplete ... hence I forked it and adjusted it to our needs.

**Comment by nicorevin** [comment]
Having compared your fork with the upstream here are the major differences:
1. No `GetCatalog` in upstream.
2. JournalEntry is a [struct](https://godoc.org/github.com/coreos/go-systemd/sdjournal#JournalEntry) with `Realtime` and `Monotonic` timestamps and `Cursor` being outside of the `Fields` map hence they don't land automatically in the event being published (should they?).
3. No `func FollowJournal(...) <-chan *sdjournal.JournalEntry`

Solution:
1. re-fork, prepare a PR and use the fork while waiting. We can turn it to an option for those who don't need it along with the `Realtime` and `Monotonic` timestamps and `Cursor` from 2.
2. add them to the resulting entry or make an option to add them if needed.
3. I've created such a func and placed it in `journalbeat/journal`. (there is an open PR and a long standing discussion about a similar feature https://github.com/coreos/go-systemd/pull/152 but it doesn't look viable. Neither know I how to fit it (and give it a proper name) to the upstream :)). Thus it will live under `journalbeat/journal` for a while.

All in all coreos lib works, I've adapted the code for the use of the upstream. I hope I didn't miss anything important while comparing. I believe if we solve 1. and 2. we will be able to the upstream flawlessly.
